hail_search/queries/mito.py

```diff
@@ -112,6 +118,215 @@ class MitoHailTableQuery(BaseHailTableQuery):
+        return self._filter_entries_ht_both_sample_types(entries_hts_map, n_partitions, **kwargs)
+
+
+    def _load_prefiltered_project_hts_both_sample_types(
+        self, project_samples: dict, n_partitions=MAX_PARTITIONS, **kwargs
+    ) -> dict[str, list[tuple[hl.Table, dict]]]:
+        entries_hts_map = {}
+        chunk_size = 64
+        project_guid_chunks: list[list[str]] = []
+        current_project_guid_chunk: list[str] = []
+        unmerged_entries_hts_map = defaultdict(dict)
+        for project_guid, project_sample_type_data in project_samples.items():
+            for sample_type, family_sample_data in project_sample_type_data.items():
+                ht = self._read_table(f'projects/{sample_type}/{project_guid}.ht',use_ssd_dir=True)
+                if ht is None:
+                    continue
+                unmerged_entries_hts_map[sample_type][project_guid] = (ht, family_sample_data)
+
+            current_project_guid_chunk.append(project_guid)
+            if len(current_project_guid_chunk) >= chunk_size:
+                project_guid_chunks.append(current_project_guid_chunk)
+                current_project_guid_chunk = []
+
+        project_guid_chunks.append(current_project_guid_chunk)
+
+        # The project tables are chunked such that for both sample types, the same variants should be in the same chunk
+        for sample_type, project_data in unmerged_entries_hts_map.items():
+            project_hts = []
+            project_family_data = {}
+            prefiltered_project_hts: list[tuple[hl.Table, dict]] = []  # Ordered list of merged, prefiltered project hts
+            for chunk in project_guid_chunks:
+                for project_guid in chunk:
+                    if project_guid not in project_data:  # There is no project ht for this sample_type
+                        continue
+                    project_ht, family_sample_data = project_data[project_guid]
+                    project_hts.append(project_ht.select_globals('sample_type', 'family_guids', 'family_samples'))
+                    project_family_data.update(family_sample_data)
+
+                self._prefilter_merged_project_hts(
+                    project_hts, project_family_data, prefiltered_project_hts, n_partitions, **kwargs
+                )
+                project_hts = []
+                project_family_data = {}
+
+            entries_hts_map[sample_type] = prefiltered_project_hts
+        return entries_hts_map
+
+    def _get_genotypes_annotation(self, include_genotype_overrides):
+        if not self._has_both_sample_types:
+            return super()._get_genotypes_annotation(include_genotype_overrides)
+
+        return {
+            GENOTYPES_FIELD: lambda r: r.family_entries.flatmap(lambda x: x).filter(
+                lambda gt: hl.is_defined(gt.individualGuid)
+            ).group_by(lambda x: x.individualGuid).map_values(lambda x: x.map(
```

Comment:
it would probably be better to only return the list if there are actually multiple sample types for that individuals (or at least for that family), so adding conditionality based on the length of `x`

Reply:
I'm trying this but running into issues with the hail conditional expression restriction that the if and else expressions must have same type. I also tried case() and this happened.